### DISCUSSION TITLE
Is it possible to intercept and prevent a route from being added via a hook?

### DISCUSSION THREAD

**Discussion by joeattardi** [fastify/fastify#4064] [Q&A] [answered]
I am working on an app that allows the user to override certain route definitions. Of course, this results in an exception because the route is already registered.

I have been able to accomplish what I need using JS proxies but it is a big mess. Was wondering if there was a way, using hooks or otherwise, to do this. 

I'd like to do something like: Before a route is registered, check if it has previously been registered; if it has, just skip it rather than throwing an error.

**Comment by mcollina** [comment, marked as answer]
Unfortunately this is not possible. You can use the `onRoute` hook to modify the route definition.

**Reply by joeattardi** [reply]
Thank you! I ended up taking a hybrid approach. I'm using the `onRoute` hook to keep track of which routes were registered, and I have a basic wrapper around a Fastify plugin that uses Proxies for the route handlers (get, post, etc.) - if it finds that the route was already registered, it will not call through.

Seems to work well enough for me!
